### ISSUE TITLE
Fedy opens with a blank window

### ISSUE THREAD

**Issue by Apopatos** [folkswithhats/fedy#557]
Hi,
just installed Fedora 27 and the first thing after that was to install Fedy.
Unfortunately it doesn't display anything. See the screenshot.

Thank you!

![2017-11-18 21-17-43](https://user-images.githubusercontent.com/25205438/32984042-23df51ee-cca7-11e7-98ef-21be2d6b5eb6.png)

**Comment by bddenhartog** [comment]
Can you paste the output of `ls -alFh /etc/yum.repos.d`?

**Comment by Apopatos** [comment]
`σύνολο 56K
drwxr-xr-x.   2 root root 4,0K Νοέ  18 21:10 ./
drwxr-xr-x. 141 root root  12K Νοέ  18 22:16 ../
-rw-r--r--.   1 root root  707 Νοέ  18 10:23 fedora-cisco-openh264.repo
-rw-r--r--.   1 root root 1,3K Οκτ  24 20:41 fedora.repo
-rw-r--r--.   1 root root  568 Νοέ  17 19:34 fedora-steam.repo
-rw-r--r--.   1 root root 1,4K Οκτ  24 20:41 fedora-updates.repo
-rw-r--r--.   1 root root 1,4K Οκτ  24 20:41 fedora-updates-testing.repo
-rw-r--r--.   1 root root  145 Νοέ  17 17:35 folkswithhats.repo
-rw-r--r--.   1 root root  173 Νοέ  17 19:09 google-chrome.repo
-rw-r--r--.   1 root root 1,1K Δεκ   6  2008 livna.repo
-rw-r--r--.   1 root root 1,1K Νοέ  18 21:10 postinstallerf.repo
`

**Comment by Apopatos** [comment]
σύνολο 56K
drwxr-xr-x.   2 root root 4,0K Νοέ  18 21:10 ./
drwxr-xr-x. 141 root root  12K Νοέ  18 22:16 ../
-rw-r--r--.   1 root root  707 Νοέ  18 10:23 fedora-cisco-openh264.repo
-rw-r--r--.   1 root root 1,3K Οκτ  24 20:41 fedora.repo
-rw-r--r--.   1 root root  568 Νοέ  17 19:34 fedora-steam.repo
-rw-r--r--.   1 root root 1,4K Οκτ  24 20:41 fedora-updates.repo
-rw-r--r--.   1 root root 1,4K Οκτ  24 20:41 fedora-updates-testing.repo
-rw-r--r--.   1 root root  145 Νοέ  17 17:35 folkswithhats.repo
-rw-r--r--.   1 root root  173 Νοέ  17 19:09 google-chrome.repo
-rw-r--r--.   1 root root 1,1K Δεκ   6  2008 livna.repo
-rw-r--r--.   1 root root 1,1K Νοέ  18 21:10 postinstallerf.repo

**Comment by bddenhartog** [comment]
Please try closing Fedy, and running the following command:

```
sudo dnf install https://download1.rpmfusion.org/free/fedora/rpmfusion-free-release-$(rpm -E %fedora).noarch.rpm https://download1.rpmfusion.org/nonfree/fedora/rpmfusion-nonfree-release-$(rpm -E %fedora).noarch.rpm
```

Then re-open Fedy. Does that work?

**Comment by Apopatos** [comment]
Still the same :-(

**Comment by bddenhartog** [comment]
@Apopatos thanks for the report. I've travel plans this weekend, but I'll take a look tomorrow and see if I can reproduce this.

**Comment by bddenhartog** [comment]
In the mean time, you can run the plugin script files directly by executing the lines you see in the `install.sh` or install line in `metadata.json` -- you can also download the scripts from Github to your local filesystem and execute them that way.

**Comment by bddenhartog** [comment]
I might ping you over the next day or two to get additional debugging information if I'm not able to recreate this right away.

**Comment by Apopatos** [comment]
Thanks a lot for your time!
I'll try your suggestions and I'll inform here.
Have a nice travel!

**Comment by Apopatos** [comment]
I downloaded the files from github and compiled them.
Now if I run the binary directly from the folder it works.
![2017-11-19 01-30-15](https://user-images.githubusercontent.com/25205438/32985713-3c1d663e-ccc9-11e7-9b88-e15fafb768d4.png)

**Comment by bddenhartog** [comment]
@Apopatos So that I can understand better, what do you mean by "compiling" Fedy?

**Comment by Apopatos** [comment]
`sudo make`
as i show in the screenshot.

**Comment by Apopatos** [comment]
OK fixed!
I removed /usr/bin/fedy and /usr/share/fedy,
installed it again and now works :-)

Thanks for your time!

**Comment by bddenhartog** [comment]
> `sudo make`
> as i show in the screenshot.

Sorry, I was commenting from mobile :)

> 
> OK fixed!
> I removed /usr/bin/fedy and /usr/share/fedy,
> installed it again and now works :-)
> 
> Thanks for your time!

Nothing has changed on our end (the package is the same), so I'd like to better understand what caused your issue in the first place. Had you initially -- meaning the very first time -- installed Fedy by cloning the repository from Github and running `sudo make`?

**Comment by Apopatos** [comment]
> Had you initially -- meaning the very first time -- installed Fedy by cloning the repository from Github and running sudo make?
Yes, that was the problem. I hadn't notice that by running make it installed the binary.

**Comment by Apopatos** [comment]
> 

>

**Comment by Apopatos** [comment]
"Had you initially -- meaning the very first time -- installed Fedy by cloning the repository from Github and running sudo make?" 
Yes, that was the problem. I hadn't notice that by running make it installed the binary.

**Comment by bddenhartog** [comment]
> > "Had you initially -- meaning the very first time -- installed Fedy by cloning the repository from Github and running sudo make?"
> Yes, that was the problem. I hadn't notice that by running make it installed the binary.

We don't ship a binary, really -- just a simple executable.

I'm still confused by what you say your initial steps that led to this issue were. If you had ran `git clone`, then `sudo make install`, you should have have a working* installation.

*You actually would be missing the application icons, since those are added via `git-lfs`.

**Comment by Apopatos** [comment]
I had ran `git clone` then `sudo make` (without install) but I hadn't notice that it installed the executable in /usr/bin so I didn't check if the installation was working. So I was running the executable directly from fedy-master folder (`./fedy`).
After that I installed Fedy via dnf and that's where the problem occured.

**Comment by gshipley** [comment]
I am having the same problem.  Fresh 27 install and ran the following:

dnf -y install https://dl.folkswithhats.org/fedora/$(rpm -E %fedora)/RPMS/folkswithhats-release.noarch.rpm

$ fedy

installed fedy-core and some other packages

fedy displays blank screen.

[zero@localhost ~]$ ls -alFh /etc/yum.repos.d
total 60K
drwxr-xr-x.   2 root root 4.0K Nov 22 11:37 ./
drwxr-xr-x. 134 root root  12K Nov 22 11:38 ../
-rw-r--r--.   1 root root  707 Oct 24 13:41 fedora-cisco-openh264.repo
-rw-r--r--.   1 root root 1.3K Oct 24 13:41 fedora.repo
-rw-r--r--.   1 root root 1.4K Oct 24 13:41 fedora-updates.repo
-rw-r--r--.   1 root root 1.4K Oct 24 13:41 fedora-updates-testing.repo
-rw-r--r--.   1 root root  145 Nov 17 10:35 folkswithhats.repo
-rw-r--r--.   1 root root 1.3K Nov 13 13:17 rpmfusion-free.repo
-rw-r--r--.   1 root root 1.3K Nov 13 13:17 rpmfusion-free-updates.repo
-rw-r--r--.   1 root root 1.3K Nov 13 13:17 rpmfusion-free-updates-testing.repo
-rw-r--r--.   1 root root 1.3K Nov 13 13:21 rpmfusion-nonfree.repo
-rw-r--r--.   1 root root 1.3K Nov 13 13:21 rpmfusion-nonfree-updates.repo
-rw-r--r--.   1 root root 1.4K Nov 13 13:21 rpmfusion-nonfree-updates-testing.repo

![screenshot from 2017-11-22 11-43-26](https://user-images.githubusercontent.com/117989/33139132-63f77286-cf7a-11e7-8999-2de5a9fece75.png)

**Comment by bddenhartog** [comment]
I'll spin up a VM and get to the bottom of this. Thanks for the reports, @gshipley and @Apopatos!

**Comment by ret2libc** [comment]
Same here. Fresh Fedora 27. Installing from the package with
```
$ dnf -y install https://dl.folkswithhats.org/fedora/$(rpm -E %fedora)/RPMS/folkswithhats-release.noarch.rpm
$ fedy
... install fedy-core and stuff...

# empty window
```

After doing
```
$ git clone https://github.com/folkswithhats/fedy
$ sudo make install
```
it works

**Comment by bddenhartog** [comment]
I've figured out the issue, but will be traveling in a few hours for half the day. I should be able to deploy a fix by Saturday morning (PST).

If anybody else is having issues, cloning from github and running `make install` will functionally work (although you will be missing some application icons if you don't use `git-lfs`).

**Comment by gshipley** [comment]
Thanks for taking a look @bddenhartog.  Much appreciated!

**Comment by webbkorey16** [comment]
I am having the same issue as @ret2libc, but the "$ sudo make install " command responds with : make: *** No rule to make target 'install'.  Stop.

**Comment by bddenhartog** [comment]
@webbkorey16 To clarify, you cloned the repository and are running into that error? That's pretty odd; the `Makefile` definitely has an `install` target.

**Comment by stoc2528** [comment]
I'm having the same problem as **webbkorey16**

*** No rule to make target 'install'.  Stop.

**Comment by harmloh** [comment]
I have same problem also , the best way to run fedy must get into make install and ./fedy 

If i run from the launcher app, it was always blank

**Comment by marcread** [comment]
had the same issue also. fedora 26.  Not sure if my comment will help much.  the cloning the git works fine. 
`git clone https://github.com/folkswithhats/fedy.git`
then run 
`./fedy/fedy`
as a hack around i juste updated the desktop launcher
 `sudo vim /usr/share/applications/org.folkswithhats.fedy.desktop `
and changed the Exec line to the one where i cloned the the git

**Comment by Olyol95** [comment]
[The issue described](https://github.com/folkswithhats/fedy/issues/557#issuecomment-346583186) by @ret2libc seems to stem from PackageKit-command-not-found suggesting the `fedy-core` package, rather than the `fedy` package. 

Using `$ sudo dnf install fedy` will install both the `fedy` and `fedy-plugins` packages that are needed to complete the installation.

**Comment by pgonzalezalv** [comment]
Just made a clean install of Fedora 27. Had the same issue (blank Fedy window) after using the command line (cf. https://www.folkswithhats.org) : 
`dnf -y install https://dl.folkswithhats.org/fedora/$(rpm -E %fedora)/RPMS/folkswithhats-release.noarch.rpm `

Issue is solved by configuring the RPMFusion with the following command line (cf. https://rpmfusion.org/Configuration) : 
`sudo dnf install https://download1.rpmfusion.org/free/fedora/rpmfusion-free-release-$(rpm -E %fedora).noarch.rpm https://download1.rpmfusion.org/nonfree/fedora/rpmfusion-nonfree-release-$(rpm -E %fedora).noarch.rpm`

**Comment by bddenhartog** [comment]
A patch should be out tomorrow. Sorry for the delay and inconvenience, things have been pretty crazy the past month.

**Comment by GabLeRoux** [comment]
Just tried again on a brand new Fedora installation

```bash
dnf -y install https://dl.folkswithhats.org/fedora/$(rpm -E %fedora)/RPMS/folkswithhats-release.noarch.rpm
```

But running `fedy` right after failed with a command not found. So I tried to run `sudo dnf install -y fedy` but it complained about a conflicting package `fedy-plugins`.

```
Error: 
Problem: package fedy-4.6.0-0.fc27.noarch requires fedy-plugins, but
none of the providers can be installed
 - conflicting requests
 - nothing provides rpmfusion-free-release needed by fedy-plugins-
4.6.0-0.fc27.noarch
```

I didn't have any `fedy` package installed (searched with `sudo dnf list installed *fedy*` and had nothing).

```
bash: fedy: command not found...
Install package 'fedy-core' to provide command 'fedy'? [N/y]
```

So I accepted and installed `fedy-core` and proceeded with changes. Then fedy opened but failed to display exactly like in current issue #557

Then ran the following as mentioned here https://github.com/folkswithhats/fedy/issues/557#issuecomment-355591251:

```bash
sudo dnf install https://download1.rpmfusion.org/free/fedora/rpmfusion-free-release-$(rpm -E %fedora).noarch.rpm https://download1.rpmfusion.org/nonfree/fedora/rpmfusion-nonfree-release-$(rpm -E %fedora).noarch.rpm
```

removed `fedy-core` with the following:

```bash
sudo dnf remove -y fedy-core
```

and installed fedy again

```bash
sudo dnf install -y fedy
```

now running `fedy` worked :v:

## tldr

I think this is what we need to do to get Fedy working on Fedora 27 in the end:

```bash
sudo dnf -y install https://dl.folkswithhats.org/fedora/$(rpm -E %fedora)/RPMS/folkswithhats-release.noarch.rpm
sudo dnf install https://download1.rpmfusion.org/free/fedora/rpmfusion-free-release-$(rpm -E %fedora).noarch.rpm https://download1.rpmfusion.org/nonfree/fedora/rpmfusion-nonfree-release-$(rpm -E %fedora).noarch.rpm
sudo dnf install -y fedy
```

**Comment by bddenhartog** [comment]
@GabLeRoux Yup, those are the correct steps. The command shown on the current iteration of the website simply installs the repository for our packages, and doesn't install RPMFusion's repositories or the `fedy` package, which provides the interface.

I've been busy and rather inactive on Github - sorry folks.
